src/CarbonAware/src/Extensions/EmissionsDataExtensions.cs

```diff
@@ -0,0 +1,83 @@
+namespace CarbonAware.Extensions;
+
+public static class EmissionsDataExtensions
+{
+    public static IEnumerable<EmissionsData> RollingAverage(this IEnumerable<EmissionsData> data, TimeSpan windowSize = default, TimeSpan tickSize = default)
+    {
+        if(data.Count() == 0){ yield break; }
+
+        if(windowSize == default)
+        {
+           foreach(var d in data){ yield return d; }
+           yield break;
+        }
+
+        var q = new Queue<EmissionsData>();
+        var _data = data.GetEnumerator();
```

Comment:
In this case, we need the enumerator anyway so I think it makes sense to use it the whole time.   Otherwise we will end up grabbing the first item and then needing to skip past it when we get the enumerator later.